src/Wallabag/CoreBundle/Controller/TagController.php

```diff
@@ -130,4 +132,46 @@ public function showEntriesForTagAction(Tag $tag, $page, Request $request)
             'tag' => $tag,
         ]);
     }
+
+    /**
+     * Rename a given tag with a new label
+     * Create a new tag with the new name and drop the old one.
+     * @param Tag     $tag
+     * @param Request $request
+     *
+     * @Route("/tag/rename/{slug}", name="tag_rename")
+     * @ParamConverter("tag", options={"mapping": {"slug": "slug"}})
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function renameTagAction(Tag $tag, Request $request)
+    {
+        $form = $this->createForm(RenameTagType::class, new Tag());
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $entries = $this->get('wallabag_core.entry_repository')->findAllByTagId(
+                $this->getUser()->getId(),
+                $tag->getId()
+            );
+            foreach ($entries as $entry) {
+                $this->get('wallabag_core.tags_assigner')->assignTagsToEntry(
+                    $entry,
+                    $form->get('label')->getData()
+                );
+            }
+
+            $em = $this->getDoctrine()->getManager();
+            $em->remove($tag);
+            $em->flush();
```

Comment:
Be careful, a tag can be associated to entries from other users.
IMHO you shouldn't delete it from the database but rather deassociate it from each entry of the current logged in user.

Reply:
Ok, I've updated the code to remove the tag from associated entries.